### PR TITLE
Add web based template dashboard

### PR DESCRIPTION
**Pull request by Drifftr**
## Purpose
This adds web based template dashboard plugin to platform. This creates an abstract dashboard and exposes an extension point(org.wso2.developerstudio.eclipse.template.dashboad.link)  to contribute to the template wizard links.

## Approach
Uses SWT.browser widget to view a web app(TemplatesDash). The JS -> Java method triggers are done through embedded server and servlets.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.templates.dashboard/TemplateDash/js/app_new.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    svgArea.animate({viewBox: (cx - 650 / 2) + ' ' + (cy - 650 / 2) + ' ' + 650 + ' ' + 650}, duration);
+}
+
+var portValue = serverDetails.port;
+
+var welcomeNodeArray;
+GetDashboardWizards();
+function loadWelcomeNodes(contributionsString) {
+    var contributions = JSON.parse(contributionsString);
+    //alert(contributionsString);
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.templates.dashboard/TemplateDash/js/app_new.js]: shall we remove those commented lines

File: plugins/org.wso2.developerstudio.eclipse.templates.dashboard/src/org/wso2/developerstudio/eclipse/templates/dashboard/web/model/DashboardLink.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.templates.dashboard/src/org/wso2/developerstudio/eclipse/templates/dashboard/web/model/DashboardLink.java]: format this file

File: plugins/org.wso2.developerstudio.eclipse.templates.dashboard/TemplateDash/js/app_new.js (hunk 1 above)
Comment (Drifftr) [reply plugins/org.wso2.developerstudio.eclipse.templates.dashboard/TemplateDash/js/app_new.js]: Fixed in d187b450db09597c86acc4bd17e64d538e412479

File: plugins/org.wso2.developerstudio.eclipse.templates.dashboard/src/org/wso2/developerstudio/eclipse/templates/dashboard/web/model/DashboardLink.java (hunk 2 above)
Comment (Drifftr) [reply plugins/org.wso2.developerstudio.eclipse.templates.dashboard/src/org/wso2/developerstudio/eclipse/templates/dashboard/web/model/DashboardLink.java]: Fixed in d187b450db09597c86acc4bd17e64d538e412479

**Pull request merged**